addOns/websocket/src/main/java/org/zaproxy/zap/extension/websocket/alerts/WebSocketAlertWrapper.java

```diff
@@ -0,0 +1,289 @@
+            return this;
+        }
+
+        public WebSocketAlertBuilder setAttack(String attack) {
+            alert.setAttack(attack);
+            return this;
+        }
+
+        public WebSocketAlertBuilder setMessage(WebSocketMessageDTO webSocketMessageDTO) {
+
+            HttpMessage handshakeMessage;
+
+            try {
+                handshakeMessage =
+                        webSocketMessageDTO.channel.getHandshakeReference().getHttpMessage();
+            } catch (Exception e) {
+                LOGGER.info("Couldn't get the Handshake Http Message for this specific channel", e);
+                return null;
+            }
+
+            alert.setMessage(handshakeMessage);
+            alert.setUri(handshakeMessage.getRequestHeader().getURI().toString());
+
+            this.webSocketMessageDTO = webSocketMessageDTO;
+            try {
+                alert.setOtherInfo(OTHER_INFO_LABEL + webSocketMessageDTO.getReadablePayload());
+            } catch (InvalidUtf8Exception e) {
+                alert.setOtherInfo(
+                        OTHER_INFO_LABEL
+                                + Constant.messages.getString("websocket.payload.invalid_utf8"));
+            }
+            return this;
+        }
+
+        public WebSocketAlertBuilder setSolution(String solution) {
+            alert.setSolution(solution);
+            return this;
+        }
+
+        public WebSocketAlertBuilder setReference(String reference) {
+            alert.setReference(reference);
+            return this;
+        }
+
+        public WebSocketAlertBuilder setEvidence(String evidence) {
+            alert.setEvidence(evidence);
+            return this;
+        }
+
+        public WebSocketAlertBuilder setCweIdm(int cweIdm) {
+            alert.setCweId(cweIdm);
+            return this;
+        }
+
+        public WebSocketAlertBuilder setWascId(int wascId) {
+            alert.setWascId(wascId);
+            return this;
+        }
+
+        public WebSocketAlertBuilder setRiskConfidence(int risk, int confidence) {
+            alert.setRiskConfidence(risk, confidence);
+            return this;
+        }
+
+        /**
+         * Sets the details of the Alert
+         *
+         * @see {@link Alert#setDetail(String, String, String, String, String, String, String,
+         *     String, int, int, HttpMessage)}
+         * @param description the description of the alert
+         * @param param the parameter that has the issue
+         * @param attack the attack triggers the issue
+         * @param webSocketMessageDTO WebSocket message triggers the issue
+         * @param solution solution for the issue
+         * @param evidence evidence (in the WebSocket message) that the issue exists
+         * @param reference references about the issue
+         * @param cweIdm the CWE ID of the issue
+         * @param wascId the WASC ID of the issue
+         */
+        public WebSocketAlertBuilder setDetails(
```

Comment:
I think we should drop this method, let the scripts/scanners use the individual setters.

Reply:
Fixed in 58cfb55f5